mingw: unrecognized options and linking error

Hi!
I'm trying to cross-compile [my project](https://github.com/4321ba/MeloMIDI/tree/main/source), including this library from ubuntu 20.04 to windows with mingw-w64 7.0.0-2, with [this](https://gist.github.com/peterspackman/8cf73f7f12ba270aa8192d6911972fe8) toolchain cmake file and I get this error:
```
x86_64-w64-mingw32-gcc: error: /arch:AVX: No such file or directory
x86_64-w64-mingw32-gcc: error: /Zi: No such file or directory

```
which I found in cmake/CXXhelpers.cmake
so my guess is that mingw doesn't know these options (and it expects dashes instead of slashes anyway)
commenting out line 18 seemed to work

however now I'm getting linking errors:
```
/usr/bin/x86_64-w64-mingw32-ld: bin/liblibnyquist.a(FlacDependencies.c.obj):FlacDependencies.c:(.text+0x1a6d2): undefined reference to `fopen_utf8'
/usr/bin/x86_64-w64-mingw32-ld: bin/liblibnyquist.a(FlacDependencies.c.obj):FlacDependencies.c:(.text+0x1a7b2): undefined reference to `fopen_utf8'
```
I actually don't need flac, nor libnyquist to manage fileio, so I just tried commenting out stuff so that it at least compiles but I just generated more errors instead. (Should I use a different library if I don't need half the stuff?)
Can someone push me in the right direction?

--- a/cmake/CXXhelpers.cmake
+++ b/cmake/CXXhelpers.cmake
@@ -14,7 +14,7 @@ function(_disable_warning flag)
 endfunction()
 
 function(_set_compile_options proj)
-    if (WIN32)
+    if(MSVC)
         target_compile_options(${proj} PRIVATE /arch:AVX /Zi )
     endif()
 endfunction()

--- a/src/FlacDependencies.c
+++ b/src/FlacDependencies.c
@@ -79,7 +79,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include "FLAC/all.h"
 
-#if defined(_MSC_VER)
+#if defined(_WIN32) || defined(_WIN64)
 #include "FLAC/src/win_utf8_io.c"
 #endif
 

